ablySpec/RealtimeClientChannel.swift

```diff
@@ -609,6 +609,65 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            // RTL8
+            context("unsubscribe") {
+
+                // RTL8a
+                it("with no arguments unsubscribes the provided listener to all messages if subscribed") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    var listener: ARTEventListener?
+                    waitUntil(timeout: testTimeout) { done in
+                        listener = channel.subscribe { message in
+                            expect(message.data as? String).to(equal("message"))
+                            done()
+                        }
+                        channel.publish(nil, data: "message")
+                    }
+
+                    channel.unsubscribe(listener!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish(nil, data: "message") { errorInfo in
+                            expect(errorInfo).to(beNil())
+                            // If `unsubscribe` fails then the test suite will raise "Done closure's was called multiple times."
+                            delay(1.0) { done() }
+                        }
+                    }
+                }
+
+                // RTL8b
+                it("with a single name argument unsubscribes the provided listener if previously subscribed with a name-specific subscription") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    var eventAListener: ARTEventListener?
+                    waitUntil(timeout: testTimeout) { done in
+                        eventAListener = channel.subscribe("eventA") { message in
+                            expect(message.data as? String).to(equal("message"))
+                            done()
+                        }
+                        channel.publish("eventA", data: "message")
+                    }
+
+                    channel.unsubscribe("eventA", listener: eventAListener!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish("eventA", data: "message") { errorInfo in
+                            expect(errorInfo).to(beNil())
+                            // If `unsubscribe` fails then the test suite will raise "Done closure's was called multiple times."
```

Comment:
As above, but also, can we please register two listeners, one for eventA and one for eventB. Then remove only eventA, and ensure that eventB still fires.  This is sensible to ensure that a unsubscribe for one listener has no side effects on other listeners.